atomisticparsers/gromacs/parser.py

```diff
@@ -83,27 +83,46 @@ def str_to_input_parameters(val_in):
                     parameters[val_array.group(1)].append(value[0] if len(value) == 1 else value)
             return parameters
 
-        def str_to_energies(val_in):
-            energy_keys_re = re.compile(r'(.+?)(?:  |\Z| P)')
-            keys = []
-            values = []
-            energies = dict()
-            for val in val_in.strip().split('\n'):
-                val = val.strip()
-                if val[0].isalpha():
-                    keys = [k.strip() for k in energy_keys_re.findall(val)]
-                    keys = ['P%s' % k if k.startswith('res') else k for k in keys if k]
+        def str_to_energies(energies_block):
+
+            max_char_name = 14
+            rows = energies_block.strip().split("\n")
+            names = [rows[i_row] for i_row in range(len(rows)) if i_row % 2 == 0]
+            values = [rows[i_row] for i_row in range(len(rows)) if i_row % 2 != 0]
+
+            names_re = r"(?<!\S)[A-Za-z0-9.()\/-]{1,}(?: [\S][A-Za-z0-9.()\/-]{1,})*(?!\S)"
+            values_re = r"[-+]?\d+\.\d+e[+-]?\d+"
+            names = [re.findall(names_re, name) for name in names]
```

Comment:
do this in a single loop

Reply:
I think this becomes moot, if you adapt my suggestion below. We should not try to do anymore regex manipulations instead we take advantage of the fixed character length for keys and values.